core/src/UVMap.cpp

```diff
@@ -138,6 +149,119 @@ auto UVMap::Plot(
     return r;
 }
 
+// pearson correlation coefficient between two vectors
+static auto pearson_correlation(
+    const std::vector<double>& x, const std::vector<double>& y) -> double
+{
+    auto n = x.size();
+    auto sum_x = std::accumulate(x.begin(), x.end(), 0.0);
+    auto sum_y = std::accumulate(y.begin(), y.end(), 0.0);
+    auto sum_xy = std::inner_product(x.begin(), x.end(), y.begin(), 0.0);
+    auto sum_x2 = std::inner_product(x.begin(), x.end(), x.begin(), 0.0);
+    auto sum_y2 = std::inner_product(y.begin(), y.end(), y.begin(), 0.0);
+
+    auto numerator = n * sum_xy - sum_x * sum_y;
+    auto denominator =
+        std::sqrt((n * sum_x2 - sum_x * sum_x) * (n * sum_y2 - sum_y * sum_y));
+
+    auto p = numerator / denominator;
+    if (std::isnan(p)) {
+        throw std::runtime_error("pearson correlation coefficient is NaN");
+    } else {
+        return p;
+    }
+}
+
+void UVMap::AlignToAxis(
+    UVMap& uv, const ITKMesh::Pointer& mesh, AlignmentAxis axis)
+{
+    // range of indices to sample from
+    std::vector<size_t> range(uv.size());
+    std::iota(range.begin(), range.end(), 0);
+
+    // number of sample points to use for alignment
+    auto num_samples = std::min(uv.size(), size_t(500));
+
+    // sample num_samples idxs
+    std::vector<size_t> idxs(num_samples);
+    std::mt19937 g(42);
+    std::sample(range.begin(), range.end(), idxs.begin(), num_samples, g);
+    std::shuffle(idxs.begin(), idxs.end(), g);
+
+    // sample UV points and corresponding mesh coordinates of interest
+    auto sampled_uvs = UVMap::New(uv);
+    sampled_uvs->clear_map();
+    std::vector<double> mesh_coords(num_samples);
+
+    for (size_t i = 0; i < num_samples; i++) {
+        auto idx = idxs[i];
+
+        sampled_uvs->set(i, uv.get(idx));
+
+        double mesh_coord = 0;
+        switch (axis) {
```

Comment:
You could move this switch outside the loop, then set `elemIdx = 0 or 1 or 2` and `elemSign = 1. or -1`. Then you can just do a one line: `meshCoord = elemSign * mesh->GetPoint(idx)[elemIdx];`. Not sure that moving the switch out of the loop makes that much performance diff, but who knows.